task/sensor/src/main.rs

```diff
@@ -14,7 +14,16 @@ use userlib::*;
 use task_sensor_api::config::NUM_SENSORS;
 
 struct ServerImpl {
-    data: &'static mut [Reading; NUM_SENSORS],
+    // We're using structure-of-arrays packing here because otherwise padding
+    // eats up a considerable amount of RAM; for example, Sidecar goes from 2868
+    // to 4200 bytes of RAM!
```

Comment:
Would it be worth hiding these behind a type that provides a getter (returning an `Option<(f32, u64)>`) / setter (taking a `f32, u64`)?

Reply:
I ended up with a slightly more complex structure-of-arrays, but I don't think getters/setter would make it more clear.  Take a look and let me know what you think.

Reply:
I think the slightly-more-complex version is actually clearer than what was there before; I agree that I don't think getters/setters would help anymore.